cobc/cobc.c

```diff
@@ -2287,7 +2287,11 @@ cobc_clean_up (const int status)
 		if (fn->need_assemble
 		 && (status
 			||  cb_compile_level > CB_LEVEL_ASSEMBLE
-			|| (cb_compile_level == CB_LEVEL_ASSEMBLE && save_temps))) {
+			|| (cb_compile_level == CB_LEVEL_ASSEMBLE
+			    && save_temps
+			    /* do not move object if a name was
+			     * specified on the command line */
+			    && !output_name))) {
```

Comment:
Does this needs to be applied to at least `-C` (translate only), too? What about `-S`?

Should it do a strcmp to output_name?

In any case: this change needs a testsuite entry.

Reply:
* `-C` and -S` work as expected in the presence of `-save-temps=DEBUG`